### PR TITLE
Ofmcc 2039 sign funding agreement

### PR DESCRIPTION
**Pull request by jenbeckett**
Sign and Submit the funding agreement. 

Please note - the Funding Agreement GUID is still hardcoded on FundingOverviewView.vue - and this will be changed upon the completion of Johnathan's work. 

Also includes ticket 2040 (update the status of FA)

### CODE REVIEW

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/views/funding/FundingConfirmationView.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    </div>
+  </v-container>
+</template>
+
+<script>
+import AppButton from '@/components/ui/AppButton.vue'
+import AppBackButton from '@/components/ui/AppBackButton.vue'
+
+export default {
+  name: 'ApplicationConfirmationView',
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/funding/FundingConfirmationView.vue:25]: done!

File: frontend/src/router/index.js [hunk 2]
```diff
@@ -141,6 +142,14 @@ const router = createRouter({
         requiresAuth: true,
       },
     },
+    {
+      path: '/funding-confirmation',
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/router/index.js:146]: Yup for sure - done :)

File: frontend/src/views/funding/FundingView.vue [hunk 3]
```diff
@@ -1,30 +1,174 @@
...
+    </v-row>
+
+    <br />
+    <br />
+    <h4 class="lg-px-10">Declaration</h4>
+    <br />
+    <br />
+
+    <v-skeleton-loader v-if="loading" :loading="loading" type="table-tbody"></v-skeleton-loader>
+    <v-row v-else class="">
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/funding/FundingView.vue:37]: done

File: frontend/src/views/funding/FundingView.vue [hunk 4]
```diff
@@ -1,30 +1,174 @@
...
+    return {
+      fundingAgreement: undefined,
+      licences: [],
+      panel: [],
+      loading: false,
+    }
+  },
+  computed: {
+    canEditFA() {
+      return this.hasPermission(this.PERMISSIONS.SIGN_FUNDING_AGREEMENT) && !this.isFundingAgreementLocked
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/funding/FundingView.vue:118]: 🤜🤛

File: frontend/src/views/funding/FundingView.vue [hunk 5]
```diff
@@ -1,30 +1,174 @@
...
+    },
+    isFundingAgreementLocked() {
+      return (
+        this.fundingAgreement?.statusCode === FUNDING_AGREEMENT_STATUS_CODES.IN_REVIEW_WITH_MINISTRY_EA ||
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/funding/FundingView.vue:122]: done!

File: frontend/src/router/index.js [hunk 6]
```diff
@@ -31,6 +31,7 @@ import ReviewApplicationView from '@/views/applications/ReviewApplicationView.vu
 import SelectFacilityView from '@/views/applications/SelectFacilityView.vue'
 import ServiceDeliveryView from '@/views/applications/ServiceDeliveryView.vue'
 import StaffingView from '@/views/applications/StaffingView.vue'
+import FundingConfirmation from '@/views/funding/FundingConfirmationView.vue'
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/router/index.js:34]: done

File: frontend/src/services/fundingAgreementService.js [hunk 7]
```diff
@@ -15,6 +15,28 @@ export default {
...
+    }
+  },
+
+  async updateFundingAgreement(fundingAgreementId, payload) {
+    try {
+      if (!fundingAgreementId) return
+      const response = await ApiService.apiAxios.patch(`${ApiRoutes.FUNDING_AGREEMENTS}/${fundingAgreementId}`, payload)
+      return response?.data
+    } catch (error) {
+      console.log(`Failed to get the list of active funding agreements by application id - ${error}`)
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/services/fundingAgreementService.js:35]: done

File: frontend/src/views/funding/FundingView.vue [hunk 8]
```diff
@@ -1,30 +1,174 @@
...
+            licence.licenceDetails = await LicenceService.getLicenceDetails(licence?.licenceId)
+          }),
+        )
+      } catch (error) {
+        this.setFailureAlert('Failed to licence(s) for facilityId = ' + this.fundingAgreement?.facilityId, error)
+      }
+    },
+
+    async submit() {
+      this.fundingAgreement.statusCode = FUNDING_AGREEMENT_STATUS_CODES.SUBMITTED
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/funding/FundingView.vue:156]: done

File: frontend/src/views/funding/FundingView.vue [hunk 9]
```diff
@@ -1,30 +1,174 @@
...
+    </v-row>
+
+    <v-row v-if="!loading" class="justify-center justify-md-space-between mx-md-7 my-3">
+      <AppBackButton id="back-home-button" width="240px" :to="{ name: 'home' }">Home</AppBackButton>
+      <AppButton
+        id="submit-funding-agreement"
+        size="large"
+        width="240px"
+        class="mt-2"
+        :disabled="isFundingAgreementLocked || !fundingAgreement.agreeConsentCertify"
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/funding/FundingView.vue:82]: done

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**